### PR TITLE
host API cleanup

### PR DESCRIPTION
**Pull request by heavenlyhash**
One tiny change suggested by @archSeer; and a rename that will make sense as the volumes API comes in next to this.

### CODE REVIEW

File: host/host.go [hunk 1]
```diff
@@ -252,7 +252,7 @@ func runDaemon(args *docopt.Args) {
 
 	// Check if we are the leader so that we can use the cluster functions directly
 	sampiCluster := sampi.NewCluster(sampi.NewState())
-	sampiAPI := sampi.NewHTTPAPI(sampiCluster)
+	sampiAPI := &sampi.HttpAPI{sampiCluster}
```
Comment (titanous) [inline host/host.go]: Vet will complain about the missing field name

File: host/http.go [hunk 2]
```diff
@@ -54,11 +54,11 @@ func (h *Host) streamEvents(id string, w http.ResponseWriter) error {
 	return nil
 }
 
-type httpAPI struct {
+type hostAPI struct {
```
Comment (heavenlyhash) [inline host/http.go]: Or actually -- maybe `jobAPI`?

Comment (archSeer) [inline host/http.go]: Sounds good

**Comment by archSeer** [comment]
LGTM, pending that rename

**Comment by archSeer** [comment]
We could also get rid of [sampi.NewCluster](https://github.com/flynn/flynn/blob/989341154e25ffde3b84437605f742bd7a04ef78/host/sampi/http.go#L21). This will need to be changed then: https://github.com/flynn/flynn/blob/35e9d7bd89f0d78a0778da92a02c47b886d97884/host/host.go#L254

**Comment by archSeer** [comment]
Ah actually, I'd change `sampi.NewCluster` to be without arguments and create a new state object within the constructor.

File: host/host.go [hunk 3]
```diff
@@ -275,16 +269,18 @@ func runDaemon(args *docopt.Args) {
...
 	for _, s := range metadata {
 		kv := strings.SplitN(s, "=", 2)
 		h.Metadata[kv[0]] = kv[1]
 	}
-	h.ID = hostID
 
+	var jobStream stream.Stream
+	sh.BeforeExit(func() {
+		if jobStream != nil {
+			jobStream.Close()
```
Comment (titanous) [inline host/host.go]: This seems like unnecessary cleanup?

File: host/sampi/http.go [hunk 4]
```diff
@@ -76,8 +76,8 @@ func (s *Cluster) StreamHostEvents(ch chan host.HostEvent, done chan bool) error
 	return nil
 }
 
-type httpAPI struct {
-	cluster *Cluster
+type HttpAPI struct {
```
Comment (titanous) [inline host/sampi/http.go]: This should be `HTTPAPI`

File: host/host.go (hunk 3 above)
Comment (archSeer) [inline host/host.go]: As I said in the commit, I found it easier to follow with the `jobStream` logic being all in one section.

Comment (titanous) [inline host/host.go]: I mean, closing the stream on exit doesn't really do much, the fds will be reaped by the kernel at exit anyway.

**Comment by titanous** [comment]
LGTM, though a few commit messages need a `pkg/` prefix.

**Pull request merged**